packages/storefront/tests/a11y/specs/accessibility.a11y.ts

```diff
@@ -52,103 +58,40 @@ test('should have successfully extracted :root styles', () => {
   expect(rootStyles).toContain('--p-animation-duration: 0s');
 });
 
-test.describe('outside main element', () => {
-  const excludeSelector = 'main';
-
-  test('should have no accessibility issues on front page', async ({ page }) => {
-    await gotoUrl(page, '/');
-    await a11yAnalyze(page, { suffix: 'light', excludeSelector });
-
-    await enableDarkMode(page);
-    await a11yAnalyze(page, { suffix: 'dark', excludeSelector });
-  });
-
-  test('should have no accessibility issues on changelog page', async ({ page }) => {
-    await gotoUrl(page, '/news/changelog'); // to have open accordion in sidebar
-    await a11yAnalyze(page, { suffix: 'light', excludeSelector });
-
-    await enableDarkMode(page);
-    await a11yAnalyze(page, { suffix: 'dark', excludeSelector });
-  });
-});
+test.describe('storefront pages', () => {
+  // filter out files from public/assets directory
+  const internalUrls = getInternalUrls().filter((url) => !url.match(/^\/assets\/.*\.\w{3,4}$/));
 
-test.describe('within main element', () => {
-  const includeSelector = 'main';
-
-  const cycleFrameworkTabs = async (page: Page, theme: string): Promise<void> => {
-    const buttons = (
-      await Promise.all([
-        // page.locator("xpath=//button[text() = 'Vanilla JS']"),
-        page.locator("xpath=//button[text() = 'Angular']"),
-        page.locator("xpath=//button[text() = 'React']"),
-        // TODO: vue is missing
-      ])
-    )
-      .map(([handle]) => handle)
-      .flat()
-      .filter(Boolean); // get rid of null values
-
-    if (buttons.length) {
-      expect(buttons.length).toBe(2);
-
-      const [vanillaJsButton] = await page.locator("//xpath=button[text() = 'Vanilla JS']");
-
-      for (const button of buttons) {
-        // pages for styles sub-package don't have vanilla js examples and angular is selected initially
-        if (vanillaJsButton) {
-          // angular and react can't be selected initially
-          expect(await button.evaluate((el: HTMLElement) => el.getAttribute('aria-selected'))).toBe('false');
+  schemes.forEach((scheme) => {
+    for (const [url, index] of internalUrls.map<[string, number]>((url, i) => [url, i])) {
+      test(`should have no accessibility issues for scheme-${scheme} at (${index + 1}/${internalUrls.length}) "${url}"`, async ({
+        page,
+        makeAxeBuilder,
+      }, testInfo) => {
+        await gotoUrl(page, url);
+
+        if (scheme === 'dark') {
+          await enableDarkMode(page);
+          // change the theme of component to dark if the option exists
+          if (await page.locator('p-select[value="light"]').first().count()) {
```

Comment:
`page.locator('p-select[value="light"]').first()` could be extracted into const, checked for existence and then used to toggle the theme. Alternatively set the theme directly in localstorage and check for the correct theme.

Reply:
Hi @henrijoss ,
the alternative with localstorage looks promising but it needs a new `BrowserContext`, or going first to another page, set the localstorage key/value and then goto the test page so that it can take the value and set the playground theme.

Reply:
I think there are ways of doing that but in my opinion not worth the effort in that case
https://github.com/microsoft/playwright/issues/6258